File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        i=0
        j=1
        max_profit=0
        n=len(prices)
        while(j<n):
            if(prices[i]>prices[j]):
                i=j
            profit=prices[j]-prices[i]
            max_profit=max(max_profit,profit)
            j+=1
        return max_profit
            

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans=0;
        int n=prices.size();
        int buy_val=prices[0];
        for(int i=1;i<n;i++){
             ans=max(ans,prices[i]-buy_val);
            if(buy_val>prices[i])
                buy_val=prices[i];
           
        }
        return ans;
    }
};